File: evidence/generated-systems/golang/manufacturing-on-golang/manufacturing-on-golang/internal/model/SalesOrderLine.go
package model

import (
    "time"
    "gorm.io/gorm"
)

//==============================================================
// SalesOrderLine Declaration
//==============================================================
type SalesOrderLine struct {
    gorm.Model
     LineNumber                                                            string
    Quantity                                                            string
    UnitPrice                                                            string
    DueDate                                                            time.Time
    SalesOrderId         *uint
    SalesOrder           *SalesOrder `gorm:"foreignKey:SalesOrderId"`
    ItemId         *uint
    Item           *Item `gorm:"foreignKey:ItemId"`

// parent associations as their child

}

File: evidence/generated-systems/django/manufacturing-on-django/manufacturingOnDjango/models/SalesOrderLine.py
from django.db import models

#======================================================================
# 
# Encapsulates data for model SalesOrderLine
#
# @author Harbormaster Dev Team
#
#======================================================================

#======================================================================
# Class SalesOrderLine Declaration
#======================================================================
class SalesOrderLine (models.Model):

#======================================================================
# attribute declarations
#======================================================================
	lineNumber = models.IntegerField(null=True)
	quantity = Quantity
	unitPrice = Money
	dueDate = models.DateField(null=True)
	salesOrder = models.ForeignKey('SalesOrder', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	item = models.ForeignKey('Item', on_delete=models.CASCADE, null=True, blank=True, related_name='+')

#======================================================================
# function declarations
#======================================================================
	def toString(self):
		str = ""
		str = str + self.lineNumber
		str = str + self.quantity
		str = str + self.unitPrice
		str = str + self.dueDate
		return str;
    
	def __str__(self):
		return self.toString();

	def identity(self):
		return "SalesOrderLine";
    
	def objectType(self):
		return "SalesOrderLine";
